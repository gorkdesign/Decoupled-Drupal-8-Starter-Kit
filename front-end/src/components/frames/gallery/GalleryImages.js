import React from 'react';
import PropTypes from 'prop-types';
// import { connect } from 'react-redux';

import Thumbnail from './Thumbnail';

export const GalleryImages = ({
  images,
  handleDelete,
}) => (
  <div className="GalleryFrameContainer">

    <div id="list" className="container p-0">
      <div className="grid">
        {
          images.map((image, i) => (<Thumbnail
            key={i}
            handleDelete={handleDelete}
            index={i}
            render={() => (
              <figure>
                <img alt="" src={image.url} className="responsive-image" />
              </figure>
              )}
          />))
        }
      </div>
    </div>
  </div>
);

GalleryImages.propTypes = {
  images: PropTypes.arrayOf(PropTypes.shape()),
  handleDelete: PropTypes.func.isRequired,
};

GalleryImages.defaultProps = {
  images: [],
};

// const mapStateToProps = (state) => {
//   const { activeArticleNid } = state.articleReducer;
//   const index = state.articleReducer.articles.findIndex(a => a.nid === activeArticleNid);
//   const { images } = state.articleReducer.articles[index];
//   return {
//     images,
//   };
// };
// const GalleryImagesWrapper = connect(mapStateToProps)(GalleryImages);

// export default GalleryImagesWrapper;

export default GalleryImages;
