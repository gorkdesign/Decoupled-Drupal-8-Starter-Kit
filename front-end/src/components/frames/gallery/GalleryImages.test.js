import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { GalleryImages } from './GalleryImages';

const initalPropState = {
  images: [],
  handleDelete: jest.fn(),
};

const generateShallow = props => shallow(<GalleryImages {...props} />);


describe('GalleryImages', () => {
  it('should render correctly', () => {
    const output = generateShallow(initalPropState);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
