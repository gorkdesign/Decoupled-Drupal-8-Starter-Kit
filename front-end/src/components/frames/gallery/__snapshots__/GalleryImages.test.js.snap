// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryImages should render correctly 1`] = `
<div
  className="GalleryFrameContainer"
>
  <div
    className="container p-0"
    id="list"
  >
    <div
      className="grid"
    />
  </div>
</div>
`;
